detect_secrets/core/secrets_collection.py

```diff
@@ -45,6 +46,20 @@ def load_from_baseline(cls, baseline: Dict[str, Any]) -> 'SecretsCollection':
     def files(self) -> Set[str]:
         return set(self.data.keys())
 
+    def scan_files(self, *filenames: str, num_processors: int = mp.cpu_count()) -> None:
+        """Just like scan_file, but optimized through parallel processing."""
+        if len(filenames) == 1:
+            self.scan_file(filenames[0])
+            return
+
+        with mp.Pool(processes=num_processors) as p:
```

Comment:
setting `processes` to `mp.cpu_count()` is probably going to be too high if running on a shared machine/CI etc. (especially if it's going to be running for several minutes clock time), maybe let this be configured?

Reply:
Added `--cores` flag to configure this.